### PR TITLE
Backend/feat/cac changes rebase

### PR DESCRIPTION
**Pull request by enigma-commits**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
CAC in nammayatri

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: lib/mobility-core/src/Kernel/External/AadhaarVerification/Types.hs [hunk 1]
```diff
@@ -13,19 +13,32 @@
...
+import qualified Kernel.Prelude as KP
 import Kernel.Storage.Esqueleto (derivePersistField)
 
 data AadhaarVerificationService = Gridline
-  deriving (Show, Read, Eq, Ord, Generic, ToJSON, FromJSON, ToSchema)
+  deriving (Show, Read, Eq, Ord, Generic, ToSchema)
+
+instance FromJSON AadhaarVerificationService where -- remove this instance once you add more constructors to AadhaarVerificationService type.
+  parseJSON (A.String val) = pure $ fromMaybe (error $ "failed to parse String " <> val <> " in AadhaarVerificationService type") (KP.readMaybe $ T.unpack val)
+  parseJSON _ = error "unexpected type, expected String for AadhaarVerificationService"
```
Comment (piyushKumar-1) [inline lib/mobility-core/src/Kernel/External/AadhaarVerification/Types.hs:38]: print the argument as well here.

File: lib/mobility-core/src/Kernel/External/Ticket/Types.hs [hunk 2]
```diff
@@ -18,13 +18,23 @@ module Kernel.External.Ticket.Types
...
+import qualified Kernel.Prelude as KP
 import Kernel.Storage.Esqueleto (derivePersistField)
 
 data IssueTicketService = Kapture
-  deriving (Show, Read, Eq, Ord, Generic, ToJSON, FromJSON, ToSchema)
+  deriving (Show, Read, Eq, Ord, Generic, ToSchema)
+
+instance FromJSON IssueTicketService where -- remove this instance once you add more constructors to IssueTicketService type.
+  parseJSON (A.String val) = pure $ fromMaybe (error $ "failed to parse String " <> val <> " in IssueTicketService type") (KP.readMaybe $ T.unpack val)
+  parseJSON _ = error "unexpected type, expected String for IssueTicketService"
```
Comment (piyushKumar-1) [inline lib/mobility-core/src/Kernel/External/Ticket/Types.hs:34]: same

File: lib/mobility-core/src/Kernel/Types/Common.hs [hunk 3]
```diff
@@ -143,7 +143,9 @@ instance FromField KVTable where
 
 data Tables = Tables
   { enableKVForWriteAlso :: [KVTable],
-    enableKVForRead :: [Text]
+    enableKVForRead :: [Text],
+    useCAC :: [Text],
```
Comment (piyushKumar-1) [inline lib/mobility-core/src/Kernel/Types/Common.hs:147]: make separate config for this.

**Review by piyushKumar-1 — CHANGES_REQUESTED** [review]

File: lib/mobility-core/src/Kernel/External/AadhaarVerification/Types.hs [hunk 4]
```diff
@@ -13,19 +13,32 @@
 -}
 {-# LANGUAGE InstanceSigs #-}
 {-# LANGUAGE TemplateHaskell #-}
+{-# OPTIONS_GHC -Wno-unrecognised-pragmas #-}
+
+{-# HLINT ignore "Use lambda-case" #-}
```
Comment (piyushKumar-1) [inline lib/mobility-core/src/Kernel/External/AadhaarVerification/Types.hs:18]: kyu ?

File: lib/mobility-core/src/Kernel/Types/Cac.hs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  { cacValue :: Value
+  }
+
+class FromJSONCAC a where
+  fromJSONCAC :: FromJSON a => CACValue a -> Result a
+
+fromJSONHelper :: FromJSON a => Value -> Maybe a
+fromJSONHelper k = case fromJSON k of
+  Success a -> Just a
+  DA.Error _ -> Nothing
```
[the remark this answers is not in the document]
Comment (enigma-commits) [reply lib/mobility-core/src/Kernel/Types/Cac.hs:21]: We simply don't care. Here In nammayatri it is handled

File: lib/mobility-core/src/Kernel/Types/Cac.hs [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+dropPrefixFromConfig :: Text.Text -> Key -> Key
+dropPrefixFromConfig key config =
```
[the remark this answers is not in the document]
Comment (enigma-commits) [reply lib/mobility-core/src/Kernel/Types/Cac.hs:24]: it is actually correct key is my key which i have to drop and config is the key coming from config.

File: lib/mobility-core/src/Kernel/Types/Cac.hs (hunk 5 above)
[the remark this answers is not in the document]
Comment (piyushKumar-1) [reply lib/mobility-core/src/Kernel/Types/Cac.hs:13]: handling is fine, i am taking about the error information is lost. But cool take it in phase 2.

File: lib/mobility-core/src/Kernel/Types/Common.hs (hunk 3 above)
Comment (piyushKumar-1) [reply lib/mobility-core/src/Kernel/Types/Common.hs:147]: phase 2

**Pull request closed without merging**